--- a/debian/additionalSrc/intellij-community/platform/util/src/com/intellij/util/AppleHiDPIScaledImage.java
+++ b/debian/additionalSrc/intellij-community/platform/util/src/com/intellij/util/AppleHiDPIScaledImage.java
@@ -1,7 +1,5 @@
 package com.intellij.util;
 
-import apple.awt.CImage;
-
 import java.awt.*;
 import java.awt.image.BufferedImage;
 
@@ -10,14 +8,10 @@
  */
 public class AppleHiDPIScaledImage {
   public static BufferedImage create(int width, int height, int imageType) {
-    return new CImage.HiDPIScaledImage(width, height, imageType) {
-      @Override
-      protected void drawIntoImage(BufferedImage image, float scale) {
-      }
-    };
+    throw new RuntimeException("apple.awt.CImage not in Debian!");
   }
 
   public static boolean is(Image image) {
-    return image instanceof CImage.HiDPIScaledImage;
+    throw new RuntimeException("apple.awt.CImage not in Debian!");
   }
 }
--- a/debian/additionalSrc/intellij-community/platform/util/src/com/intellij/util/ui/IsRetina.java
+++ b/debian/additionalSrc/intellij-community/platform/util/src/com/intellij/util/ui/IsRetina.java
@@ -15,25 +15,10 @@
  */
 package com.intellij.util.ui;
 
-import apple.awt.CImage;
-
 import java.awt.image.BufferedImage;
 
 class IsRetina {
   public static boolean isRetina() {
-    try {
-      final boolean[] isRetina = new boolean[1];
-      new CImage.HiDPIScaledImage(1, 1, BufferedImage.TYPE_INT_ARGB) {
-        @Override
-        public void drawIntoImage(BufferedImage image, float v) {
-          isRetina[0] = v > 1;
-        }
-      };
-      return isRetina[0];
-    }
-    catch (Throwable e) {
-      e.printStackTrace();
-      return false;
-    }
+    throw new RuntimeException("apple.awt.CImage not in Debian!");
   }
 }
