--- a/build-system/gradle-core/build.gradle
+++ b/build-system/gradle-core/build.gradle
@@ -15,6 +15,7 @@
 sourceSets {
     main {
         groovy.srcDirs = ['src/main/groovy', 'src/fromGradle/groovy']
+        groovy.exclude '**/JacocoTransform.java'
         resources.srcDirs = ['src/main/resources', 'src/fromGradle/resources']
         compileClasspath += configurations.provided
     }
@@ -37,7 +38,7 @@
     compile 'org.ow2.asm:asm:5.0.3'
     compile 'org.ow2.asm:asm-commons:5.0.3'
     compile "net.sf.proguard:proguard-gradle:${project.ext.proguardVersion}"
-    compile "org.jacoco:org.jacoco.core:0.7.4.201502262128"
+    //compile "org.jacoco:org.jacoco.core:0.7.4.201502262128"
     //testCompile project(':base:instant-run:instant-run-annotations')
     //testCompile project(':base:instant-run:instant-run-runtime')
 
--- a/build-system/gradle-core/src/main/groovy/com/android/build/gradle/internal/TaskManager.java
+++ b/build-system/gradle-core/src/main/groovy/com/android/build/gradle/internal/TaskManager.java
@@ -96,7 +96,6 @@
 import com.android.build.gradle.internal.transforms.InstantRunSlicer;
 import com.android.build.gradle.internal.transforms.InstantRunTransform;
 import com.android.build.gradle.internal.transforms.InstantRunVerifierTransform;
-import com.android.build.gradle.internal.transforms.JacocoTransform;
 import com.android.build.gradle.internal.transforms.JarMergingTransform;
 import com.android.build.gradle.internal.transforms.NoChangesVerifierTransform;
 import com.android.build.gradle.internal.transforms.MergeJavaResourcesTransform;
@@ -2291,12 +2290,7 @@
     public void createJacocoTransform(
             @NonNull TaskFactory taskFactory,
             @NonNull final VariantScope variantScope) {
-
-        AndroidTask<?> task = variantScope.getTransformManager().addTransform(taskFactory,
-                variantScope, new JacocoTransform(project.getConfigurations()));
-
-        Copy agentTask = getJacocoAgentTask();
-        task.dependsOn(taskFactory, agentTask);
+        throw new UnsupportedOperationException("JaCoCo is not in Debian yet.");
     }
 
     public void createJackTask(
